net-p2p/rtorrent: version bump 0.9.6

anthonyryan1 opened the pull request:
* Bumped net-libs/libtorrent as well since those two packages are released together
* Updated the HOMEPAGE and SRC_URI due to changes upstream
* [Repoman] Bumped EAPI, copyright and added slot operators
* Added additional metadata and fixed indentation of metadata.xml

Fixes https://bugs.gentoo.org/show_bug.cgi?id=562830
Additionally this release fixes the broken unit tests that prevented 0.9.4 from ever being stabilized

mgorny commented on net-libs/libtorrent/libtorrent-0.13.6.ebuild:
```diff
@@ -0,0 +1,48 @@
+	virtual/pkgconfig"
+
+src_prepare() {
+#	epatch "${FILESDIR}"/download_constructor.diff
```
Any reason to keep it? I guess it's fixed upstream, correct?

anthonyryan1 replied:
It's hard to tell definitively if this was fixed upstream. That patch references a bug number on their old bug tracker which was not archived when they moved everything to a new location. Making it hard to see how that issue was ultimately resolved.

With that said, the patch says it's intended as a quick and dirty fix before calling `parse_tracker()`, and that call has since been removed from this bit of code. This makes me think it's safe to remove.

The uncertainty over whether it was actually fixed is likely why it was left in the previous ebuild commented out as well. We're not going to get any more certain in the future so I will remove it.

mgorny commented on net-libs/libtorrent/libtorrent-0.13.6.ebuild:
```diff
@@ -0,0 +1,48 @@
+	# the configure check for posix_fallocate is wrong.
+	# reported upstream as Ticket 2416.
+	local myconf
+	echo "int main(){return posix_fallocate();}" > "${T}"/posix_fallocate.c
```
Could you add `||die` here, please? At the end of the line.

anthonyryan1 replied:
It looks to me as though this whole section can be removed instead. Again it's not possible to reference the upstream bug with the bug tracker being gone but it is correctly detecting the presence of `posix_fallocate()`.

I'll set up a chroot and try to confirm it will work without `posix_fallocate()` available as well, then we may be able to simplify further here.

mgorny commented on net-libs/libtorrent/libtorrent-0.13.6.ebuild:
```diff
@@ -0,0 +1,48 @@
+
+EAPI=5
+
+inherit eutils libtool toolchain-funcs
```
Unless I'm missing something, eutils is unused now.

hasufell replied:
now it is used again :D

mgorny commented on net-p2p/rtorrent/rtorrent-0.9.6.ebuild:
```diff
@@ -0,0 +1,63 @@
+		"${FILESDIR}"/${P}-tinfo.patch
+
+	# upstream forgot to include
+	cp "${FILESDIR}"/rtorrent.1 "${S}"/doc/ || die
```
Is this still a case?

mgorny replied:
Yep, looks like it. I wonder if anyone bothered to report this…

anthonyryan1 replied:
The man page is available at `doc/old/rtorrent.1`. I'll remove this and change the `doman`, but `files/rtorrent.1` will need to stay until the older ebuilds are removed.

anthonyryan1 replied:
It seems I spoke too soon on this one.

While it is in the source tree at this location. It seems they're stripping it out of the official release tarballs. If we wanted to start fetching the tag tarballs off of github we can use that location. But if we want to stick with the official upstream release tarball, we cannot.

Since I think we should probably stick with the tarballs linked from the homepage, it seems this is going to need to continue for now.

anthonyryan1 commented:
Pull request updated per comments so far.

* Added missing test dependency for libtorrent
* Removed posix_falloc test override (resolved upstream)
* `${FILESDIR}"/download_constructor.diff` has been removed. It is still referenced in comments of older ebuilds, but not used.

I welcome further comments and criticisms.

mgorny commented:
Only one more: the commit message is too wide. Please wrap at 72 or 80 characters ;-). You may also want to PGP-sign the commit, https://wiki.gentoo.org/wiki/Project:Gentoo-keys/Generating_GLEP_63_based_OpenPGP_keys.

hasufell commented on net-libs/libtorrent/libtorrent-0.13.6.ebuild:
```diff
@@ -0,0 +1,38 @@
+src_configure() {
+	# configure needs bash or script bombs out on some null shift, bug #291229
+	CONFIG_SHELL=${BASH} econf \
+		--disable-dependency-tracking \
```
this is already default since EAPI=4

hasufell commented on net-libs/libtorrent/libtorrent-0.13.6.ebuild:
```diff
@@ -0,0 +1,38 @@
+SRC_URI="http://rtorrent.net/downloads/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
```
how about we add subslot for the soname? It appears to be `libtorrent.so.19` so we could do `SLOT=0/19`

anthonyryan1 replied:
The reason this hasn't been done in the past is because the only program which uses this is `net-p2p/rtorrent` and each version of rtorrent is only usable with exactly one version of libtorrent that was released together with it. Because of that, even if the soname didn't change, the resulting program would be buggy if the rtorrent and libtorrent versions didn't match up correctly.

I'm fairly confident if we add a subslot now, in the future someone will see `net-p2p/rtorrent` using a version instead of a subslot, correct it and start introducing subtle bugs.

hasufell replied:
that indicates the soname is broken and doesn't reflect ABI

hasufell commented on net-libs/libtorrent/libtorrent-0.13.6.ebuild:
```diff
@@ -0,0 +1,38 @@
+		$(use_enable ipv6) \
+		$(use_enable ssl openssl) \
+		--with-posix-fallocate
+}
```
I get a seemingly useless `/usr/lib64/libtorrent.la`, so maybe we want to run `prune_libtool_files` from eutils.eclass?

anthonyryan1 replied:
Fixed in `src_install`.

hasufell commented on net-libs/libtorrent/libtorrent-0.13.6.ebuild:
```diff
@@ -0,0 +1,38 @@
+
+RDEPEND="
+	>=dev-libs/libsigc++-2.2.2:2
+	ssl? ( dev-libs/openssl:0= )"
```
it compiles with libressl, so make this
```
ssl? (
    !libressl? ( dev-libs/openssl:0= )
    libressl? ( dev-libs/libressl:= )
)
```
and add `libressl` to `IUSE`

hasufell commented on net-libs/libtorrent/libtorrent-0.13.6.ebuild:
```diff
@@ -0,0 +1,38 @@
+KEYWORDS="~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86 ~x86-fbsd ~amd64-linux ~arm-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~sparc-solaris ~x64-solaris"
+IUSE="debug ipv6 ssl test"
+
+RDEPEND="
```
I get explicit linkage to `sys-libs/zlib` and `src/torrent/utils/log.cc` contains `#include <zlib.h>`

anthonyryan1 commented:
Commit text has been limited, I'll see about getting a ruler by default for future commit messages.

I've also signed the commit, with the disclosure that I've never signed a commit before so if there's any problems with the signature, please let me know.

perfinion commented:
@anthonyryan1 I tested this and it looks good to me. Can you rebase it on the current HEAD then ping me and i'll merge it in. It is pretty far back so it'll be a bit cleaner.

perfinion commented:
@anthonyryan1 actually, can you change this line too:
Fixes https://bugs.gentoo.org/show_bug.cgi?id=562830
change it to the shorter url:
Fixes https://bugs.gentoo.org/562830

anthonyryan1 commented:
@perfinion Rebased, bug url shortened.

anthonyryan1 commented:
After checking that QA failure, I don't believe it relates to this set of changes. Please feel free to ping me if I'm overlooking something.

perfinion commented:
@anthonyryan1 the QA error looks like some other package happened to be broken in the tree at the same time the checks were run on your package so its fine.

I have merged the pull request in. Thanks for the work :D
